### PR TITLE
Add API for unban

### LINKED ISSUE #7: No error body when request target field is array
If we request api with body like this:
`{
  "target": [
    {
      "ip": "192.168.10.123"
    }
  ]
}`
response had a 400 Bad Request status code, but response body is empty.

### PR DESCRIPTION
**Pull request by horny-manul**
Closes #8 
Closes #7

### CODE REVIEW

File: src/unban/service.rs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        let cli = reqwest::Client::new();
+        Service { cli, executors }
+    }
+}
+
+#[async_trait]
+impl UnBanner for Service {
+    async fn unban(&self, ur: UnBanRequest) -> Result<(), Vec<UnbanStatus>> {
+        let mut ubs = Vec::new();
+        for exec in &self.executors {
```
Comment (stuffy-idler) [inline src/unban/service.rs:24]: I suggest to rename `exec` to `executor` – `exec` is commonly used name for function, and saving 4 symbols in the variable name will worsen readability

Comment (stuffy-idler) [inline src/unban/service.rs:24]: It's better to rewrite `for - in` to combination of `map` that will produce vec of futures and `futures::future::join_all` that will executor all of them concurrently and return ordered vec of outputs

File: src/model.rs [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
+#[serde(rename_all = "snake_case")]
+#[serde(untagged)]
+pub enum UnBanEntity {
+    Target(BanTarget),
+    Pattern(String),
+}
+
+impl Display for UnBanEntity {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.write_str(serde_json::to_string(self).unwrap().as_str())
```
Comment (stuffy-idler) [inline src/model.rs:14]: `unwrap` usage is dangerous for runtime. Moreover, inside a function that returns `Result` it should be replaced with `Err`

File: src/main.rs [hunk 3]
```diff
@@ -30,17 +31,14 @@ async fn main() -> anyhow::Result<()> {
     };
 
     let dur: std::time::Duration = cfg.redis_query_timeout.into();
```
Comment (stuffy-idler) [inline src/main.rs:33]: `dur` is not descriptive name, change it

File: src/main.rs [hunk 4]
```diff
@@ -5,6 +5,7 @@ mod error;
 mod model;
 mod redis;
 mod telemetry;
+mod unban;
```
Comment (stuffy-idler) [inline src/main.rs:8]: I suggest to name it similarly to `ban_checker`, e.g. `unbanner`

**Review by stuffy-idler — CHANGES_REQUESTED** [review]

File: src/error.rs [hunk 5]
```diff
@@ -35,3 +37,22 @@ pub enum Redis {
     #[error("timeout")]
     Timeout,
 }
+
+#[derive(Debug, PartialEq)]
+pub enum BanTargetConversionError {
+    IPOrUserAgentRequired,
```
Comment (stuffy-idler) [inline src/error.rs:43]: It's not extendable error name, I suggest to name it `MissingField` or something similar

File: src/api/routes/unban.rs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+) -> Result<impl Responder, impl ResponseError> {
+    if let Err(e) = unban_req.verify() {
+        return Err(e.into());
+    }
+
+    match unban.unban(unban_req.0).await {
+        Ok(()) => Ok(HttpResponse::NoContent().finish()),
+        Err(e) => {
+            tracing::error!("ban target: {:?}", e);
+            let err_resp: ErrorResponse = e.into();
```
Comment (stuffy-idler) [inline src/api/routes/unban.rs:39]: I guess it can be done without extra variable

**Review by stuffy-idler — CHANGES_REQUESTED** [review]

File: src/unban/service.rs [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+            let resp = self
+                .cli
+                .delete(&exec.url)
+                .body(serde_json::to_vec(&ur).expect("UnBanEntity derives Serialize"))
```
[the remark this answers is not in the document]
Comment (horny-manul) [reply src/unban/service.rs]: done here https://github.com/crypto-pepe/firewall-api/pull/12/files#diff-2fc5713c1adda4e554a333bc88f0c0781d144c34ea47d86c3f9209123e2bfd79R56

File: src/api/routes/unban.rs [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+use crate::unbanner::UnBanner;
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct UnBanRequest {
+    pub target: Option<UnBanEntity>,
+}
+
+impl UnBanRequest {
+    pub fn verify(&self) -> Result<(), BanTargetConversionError> {
+        if self.target.is_none() {
```
[the remark this answers is not in the document]
Comment (horny-manul) [reply src/api/routes/unban.rs:17]: done here https://github.com/crypto-pepe/firewall-api/pull/12/files#diff-c1591b764f87289591af441242f855ede3b0b57d71267f9a2beece30697c11d1R17

File: src/api/routes/unban.rs (hunk 6 above)
Comment (horny-manul) [reply src/api/routes/unban.rs:39]: done here https://github.com/crypto-pepe/firewall-api/pull/13/files#diff-a4487fc58e096dc35553836b2e679c57256578af38b1e97e2ae83ece311e0652R27

**Pull request closed without merging**